executor/android/android_seccomp.h

```diff
@@ -0,0 +1,318 @@
+	}
+}
+
+void _install_setuidgid_filter(uint32 uid_gid_min, uint32 uid_gid_max)
+{
+	filter f;
+#ifdef DUAL_ARCH
+	// Note that for mixed 64/32 bit architectures, ValidateArchitecture inserts a
+	// jump that must be changed to point to the start of the 32-bit policy
+	// 32 bit syscalls will not hit the policy between here and the call to SetJump
+	auto offset_to_secondary_filter = ValidateArchitectureAndJumpIfNeeded(f);
+#else
+	ValidateArchitecture(f);
+#endif
+
+	ValidateSetUidGid(f, uid_gid_min, uid_gid_max, true /* primary */);
+
+#ifdef DUAL_ARCH
+	// Will fail() if anything fails.
+	SetValidateArchitectureJumpTarget(offset_to_secondary_filter, f);
+
+	ValidateSetUidGid(f, uid_gid_min, uid_gid_max, false /* primary */);
+#endif
+
+	// Will fail() if anything fails.
+	install_filter(f);
+}
+
+enum FilterType {
+	APP,
+	APP_ZYGOTE,
+	SYSTEM,
+};
+
+void _set_seccomp_filter(FilterType type)
+{
+	const sock_filter *p, *s;
+	size_t p_size, s_size;
+	filter f;
+
+	switch (type) {
+	case APP:
+		p = primary_app_filter;
+		p_size = primary_app_filter_size;
+		s = secondary_app_filter;
+		s_size = secondary_app_filter_size;
+		break;
+	case APP_ZYGOTE:
+		p = primary_app_zygote_filter;
+		p_size = primary_app_zygote_filter_size;
+		s = secondary_app_zygote_filter;
+		s_size = secondary_app_zygote_filter_size;
+		break;
+	case SYSTEM:
+		p = primary_system_filter;
+		p_size = primary_system_filter_size;
+		s = secondary_system_filter;
+		s_size = secondary_system_filter_size;
+		break;
+	default:
+		fail("Type is wrong %d", type);
+	}
+
+#ifdef DUAL_ARCH
+	// Note that for mixed 64/32 bit architectures, ValidateArchitecture inserts a
+	// jump that must be changed to point to the start of the 32-bit policy
+	// 32 bit syscalls will not hit the policy between here and the call to SetJump
+	auto offset_to_secondary_filter = ValidateArchitectureAndJumpIfNeeded(f);
+#else
+	ValidateArchitecture(f);
+#endif
+
+	ExamineSyscall(f);
+
+	for (size_t i = 0; i < p_size; ++i) {
+		f.push_back(p[i]);
+	}
+	Disallow(f);
+
+#ifdef DUAL_ARCH
+	// Will fail() if anything fails.
+	SetValidateArchitectureJumpTarget(offset_to_secondary_filter, f);
+
+	ExamineSyscall(f);
+
+	for (size_t i = 0; i < s_size; ++i) {
+		f.push_back(s[i]);
+	}
+	Disallow(f);
+#endif
+
+	// Will fail() if anything fails.
+	install_filter(f);
+}
+
+// Modified from the orignal Android code to fail instead of return.
+void set_app_seccomp_filter()
+{
+	_set_seccomp_filter(FilterType::APP);
+}
+
+// Modified from the orignal Android code to fail instead of return.
+void set_app_zygote_seccomp_filter()
+{
+	_set_seccomp_filter(FilterType::APP_ZYGOTE);
+}
+
+// Modified from the orignal Android code to fail instead of return.
+void set_system_seccomp_filter()
+{
+
+	_set_seccomp_filter(FilterType::SYSTEM);
+}
+
+// Modified from the orignal Android code to fail instead of return.
+void install_setuidgid_seccomp_filter(uint32 uid_gid_min, uint32 uid_gid_max)
```

Comment:
Is there plan to use filters other than set_app_seccomp_filter?
This adds dead code to the project and to every reproducer and this _install_setuidgid_filter also pulls in syscall number tables that don't seem to be necessary otherwise. Also complicates/delays review.
I would prefer if this is added later when needed, until you have next PR ready that makes use of them.

Reply:
Lets drop it for now